python/brayns/client.py

```diff
@@ -0,0 +1,226 @@
+
+    # pylint: disable=W0613,W0622,E1101
+    def image(self, size, format='jpg', quality=None, samples_per_pixel=None):
+        """
+        Request a snapshot from Brayns and return a PIL image.
+
+        :param tuple size: (width,height) of the resulting image
+        :param str format: image type as recognized by FreeImage
+        :param int quality: compression quality between 1 (worst) and 100 (best)
+        :param int samples_per_pixel: samples per pixel to increase render quality
+        :return: the PIL image of the current rendering, None on error obtaining the image
+        :rtype: :py:class:`~PIL.Image.Image`
+        """
+        args = locals()
+        del args['self']
+        result = self.snapshot(response_timeout=None, **{k: v for k, v in args.items() if v})
+
+        # error case: invalid request/parameters
+        if 'code' in result:
+            print(result['message'])
+            return None
+
+        # https://stackoverflow.com/a/9807138
+        data = result['data']
+        missing_padding = len(data) % 4
+        if missing_padding != 0:
+            data += b'=' * (4 - missing_padding)
+        return Image.open(io.BytesIO(base64.b64decode(data)))
+
+    def set_colormap(self, colormap='magma', colormap_size=256, intensity=1, opacity=1,
+                     data_range=(0, 256)):
+        """
+        Set a colormap to Brayns.
+
+        :param str colormap: color palette to use from matplotlib and seaborn
+        :param int colormap_size: the number of colors to use to control precision
+        :param int intensity: value to amplify the color values
+        :param int opacity: opacity for colormap values
+        :param typle data_range: data range on which values the colormap should be applied
+        """
+        import seaborn as sns
+        palette = sns.color_palette(colormap, colormap_size)
+        palette_size = len(palette)
+        contributions = []
+        diffuses = []
+        # pylint: disable=E1101
+        tf = self.transfer_function
+        for i in range(0, palette_size):
+            color = palette[i]
+            diffuses.append([intensity * color[0], intensity * color[1], intensity * color[2],
+                             opacity])
+            contributions.append(0)
+        tf.diffuse = diffuses
+        tf.contribution = contributions
+        tf.range = data_range
+        tf.commit()
+
+    def open_ui(self):
+        """Open the Brayns UI in a new page of the default system browser."""
+        import webbrowser
+        url = 'https://bbp-brayns.epfl.ch?host=' + self.url()
+        webbrowser.open(url)
+
+    def _check_version(self):
+        """Check if the Brayns' version is sufficient enough."""
+        status = utils.http_request(HTTP_METHOD_GET, self.url(), 'version')
+        if status.code != HTTP_STATUS_OK:
+            raise Exception('Cannot obtain version from Brayns')
+
+        minimal_version = '0.7.0'
```

Comment:
Maybe we could keep this min version in the same version file where we have the python version for the lib?